### PR TITLE
Enable `no_std` support via a default-enabled `std` feature flag.

### PR DESCRIPTION
**Pull request by nuttycom**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "redjubjub"
-edition = "2018"
+edition = "2021"
```
Comment (nuttycom) [inline Cargo.toml:3]: This change does not change the MSRV, as `reddsa 0.5.0` already requires `1.65`.

File: Cargo.toml [hunk 2]
```diff
@@ -17,10 +17,10 @@ description = "A standalone implementation of the RedJubjub signature scheme."
 features = ["nightly"]
 
 [dependencies]
-reddsa = "0.5.0"
-rand_core = "0.6"
+reddsa = { version = "0.5.0", default-features = false, features = ["alloc"] }
+rand_core = { version = "0.6", default-features = false }
 serde = { version = "1", optional = true, features = ["derive"] }
```
Comment (str4d) [inline Cargo.toml:22]: This should also have `default-features = false` to ensure that enabling the `serde` feature flag doesn't force `std` to also be enabled.

**Review by str4d — CHANGES_REQUESTED** [review]
Reviewed 96c0b799b72d324b35afb1ea90cdc52150e29ce8.

**Review by str4d — APPROVED** [review]
utACK e413019904400f4caa3550df7c4040befadfbb14

**Review by conradoplg — APPROVED** [review]

**Pull request merged**